src/modm/platform/uart/stm32/uart.hpp.in

```diff
@@ -49,21 +52,27 @@ public:
 	// Expose jinja template parameters to be checked by e.g. drivers or application
 	static constexpr size_t RxBufferSize = {{ options["buffer.rx"] }};
 	static constexpr size_t TxBufferSize = {{ options["buffer.tx"] }};
+%% if options["buffer.rx"]
+	static inline modm::atomic::Queue<uint8_t, {{ options["buffer.rx"] }}> rxBuffer;
+%% endif
+%% if options["buffer.tx"]
+	static inline modm::atomic::Queue<uint8_t, {{ options["buffer.tx"] }}> txBuffer;
```

Comment:
At this point I'd make this a templated class so that we can remove the `buffer.*` options. I'm ok with breaking changes here, this is a significant improvement.

Reply:
Actually if we're gonna do breaking changes we can also deduplicate the common code as you initially intended.

Reply:
If I don't need to keep it backward-compatible then it becomes much easier.  I'll remove the `buffer.*` options and will do it via another template specialisation which uses `modm::atomic::Queue`

Reply:
I deleted the `buffer.*` options (and the old `Uart` class), which obviously broke a lot of examples. I fixed the `blue_pill_f103` examples, it was just a matter of adding 

```using Usart2 = BufferedUart<UsartHal2, PlainTxBuffer<256>>;```

at the top of main.cpp. Before I go ahead and convert all other examples - are you happy with this?

Reply:
Yes, I think this is pretty awesome, very flexible and fullfills all requirements. The recursive callback chaining is also quite elegant.